[Communication] Adding ACS NetworkTraversal 2022-03-01-preview API

ajpeacock0 opened the pull request:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

This swagger introduces an optional request parameter to the ACS NetworkTraversal API allowing users to specify the TTL for TURN credentials when fetching a Relay Configuration.

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [X] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  2022-02-30
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
 2022-03-01
  4. If updating an existing version, please select the specific langauge SDKs and CLIs that must be refreshed after the swagger is published.
      - [x] SDK of .NET (need service team to ensure code readiness)
      - [x] SDK of Python
      - [x] SDK of Java
      - [x] SDK of Js
      
### Contribution checklist:
- [X] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [X] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [X] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

mikekistler commented on specification/communication/data-plane/NetworkTraversal/stable/2022-03-01/CommunicationNetworkTraversal.json:
```diff
@@ -0,0 +1,179 @@
+        "ttl": {
+          "type": "integer",
+          "format": "int32",
+          "description": "The credential Time-To-Live (TTL), in seconds. The default value (172800) will be used if given value exceeds it."
```
You can specify the default value in OpenAPI with the "default" property:
```suggestion
          "description": "The credential Time-To-Live (TTL), in seconds. The default value (172800) will be used if given value exceeds it.",
          "default": 172800
```

mikekistler reviewed (approved):
This looks good, but in the future please start with a commit that simply copies the prior version and then make your changes in a new commit on top of that -- to make it clear what is being changed (and not changed) in the PR.

I also left a suggestion for adding the default explicitly but that's just a recommendation.

mikekistler commented on specification/communication/data-plane/NetworkTraversal/stable/2022-03-01/CommunicationNetworkTraversal.json:
```diff
@@ -0,0 +1,179 @@
+      "name": "api-version",
+      "description": "Version of API to invoke.",
+      "required": true,
+      "enum": [
+        "2022-03-01"
+      ],
```
Please remove the enum from this definition or it will cause you perpetual grief.

jorgegarchirota replied:
QQ, in the past reviews, the board suggested to add the parameter, so, what it would be the best practice here?

ajpeacock0 replied:
I found that I hit an issue last year when changing a swagger path. The [1005 rule ](https://github.com/Azure/openapi-diff/blob/master/docs/rules/1005.md) I was breaking stated

> Description: Checks whether an existing path is removed from the previous specification.
> Cause: This is considered a breaking change. This change requires new api-version.

and I reached out to Joel Hendrix about why my change, which contained a new API version, appeared to be getting this error.

Joel told me this is because the API version wasn't actually specified in my ApiVersionParameter param (in either swagger) so the comparison thinks they're the same version and I should add it to the enum like this https://github.com/Azure/azure-rest-api-specs/blob/master/specification/cognitiveservices/data-plane/TextAnalytics/stable/v3.1/TextAnalytics.json#L2667

This resulted in me adding the enum to the ApiVersionParameter for the previous swaggers: https://github.com/Azure/azure-rest-api-specs/pull/14955/files

However not it seems that I shouldn't be including this enum?

CC: @DominikMe

ajpeacock0 replied:
After discussion, it was decided that this enum should be removed

mikekistler reviewed (changes requested):
I missed the enum in the api-version.  This is what the pipeline is complaining about. You need to remove that. This may still need a breaking change review/approval, but once removed you will able to avoid that going forward.

mikekistler commented:
And you need to fix the [Avocado error](https://github.com/Azure/azure-rest-api-specs/pull/17502/checks?check_run_id=4943845770), which is basically saying that you need to update the [readme.md](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/communication/data-plane/NetworkTraversal/readme.md) to add a tag for your new api-version, pointing to the new version of the API defn.

mikekistler reviewed (approved):
Looks good! 👍

jhendrixMSFT commented:
/azp run

ajpeacock0 committed:
Adding ACS NetworkTraversal 2022-03-01 API

ajpeacock0 commented:
@tjprescott I have removed the "PENDING DEPLOYMENT" tag from the title, since this swagger is now ready to be merged. Side note, related to the recent GA noise, I changed the API version to include the `-preview`
